container/monitor.go

```diff
@@ -304,8 +304,7 @@ func (m *containerMonitor) shouldRestart(exitCode int) bool {
 // received ack from the execution drivers
 func (m *containerMonitor) callback(processConfig *execdriver.ProcessConfig, pid int, chOOM <-chan struct{}) error {
 	go func() {
-		_, ok := <-chOOM
-		if ok {
+		for range chOOM {
```

Comment:
The channel from libcontainer's `NotifyOOM` is reliably closed afaict, and in every other case an already closed channel is passed in.

The loop should end as soon as the channel is closed.

In addition, if I run a build of this, while a container is running the SIGUSR1 output indicates one goroutine at `monitor.go:307`, and then after stopping the container (with OOM or otherwise), the SIGUSR1 output shows no goroutines in `monitor.go` at all.